sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroid.java

```diff
@@ -104,20 +104,29 @@ public static synchronized void init(
 
             final BuildInfoProvider buildInfoProvider = new BuildInfoProvider(logger);
             final LoadClass loadClass = new LoadClass();
+            final ActivityFramesTracker activityFramesTracker =
+                new ActivityFramesTracker(loadClass, options);
 
             AndroidOptionsInitializer.loadDefaultAndMetadataOptions(
                 options, context, logger, buildInfoProvider);
 
-            configuration.configure(options);
-
-            AndroidOptionsInitializer.initializeIntegrationsAndProcessors(
-                options,
+            // We install the default integrations before the option configuration, so that the user
+            // can remove any of them. Integrations will not evaluate the options immediately, but
+            // will use them later, after being configured.
+            AndroidOptionsInitializer.installDefaultIntegrations(
```

Comment:
I think moving this above `configuration.configure()` will break what I fixed here a while ago: https://github.com/getsentry/sentry-java/pull/2377
E.g. `AndroidOptionsInitializer.installDefaultIntegrations` calls `AndroidEnvelopeCache.hasStartupCrashMarker(options)` which itself calls `options.getOutboxPath()`, 	which itself checks a folder based on the DSN - which might not be set at this point in time.

Reply:
I added `LazyEvaluator` class to evaluate a function lazily
`AndroidOptionsInitializer.installDefaultIntegrations` now evaluates the cache dir lazily, so that should be covered